### PR TITLE
feat(avatar): add Avatar component

### PR DESCRIPTION
**Pull request by liuderchi**
## [](#what-i-did)What I did

- add `<Avatar>` component for displaying user avatar
- add svg image for default condition; render `/avatar.svg` with `svgr.macro`
- add storybook examples for general case and default case

## [](#how-to-test)How to test

1. `yarn start`
2. checkout `Avatar` section in localhost:6006

### CODE REVIEW

**Comment by evenchange4** [comment]
Deploy preview for *mcs-lite-ui* ready!

Built with commit 1c6e66b7a23a9a31272e8b7421544c325fd62435

https://deploy-preview-602--mcs-lite-ui.netlify.com

**Comment by evenchange4** [comment]
Deploy preview for *mcslite* ready!

Built with commit 1c6e66b7a23a9a31272e8b7421544c325fd62435

https://deploy-preview-602--mcslite.netlify.com

**Comment by evenchange4** [comment]
Deploy preview for *mcs-lite-introduction* ready!

Built with commit 1c6e66b7a23a9a31272e8b7421544c325fd62435

https://deploy-preview-602--mcs-lite-introduction.netlify.com

File: packages/mcs-lite-ui/src/Avatar/Avatar.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { compose, pure } from 'recompose';
+import IconAvatar from './IconAvatar';
+import Wrapper from './styled-components';
+
+type Props = {
+  src?: string,
+  size?: number,
+};
+
+const PureAvatar = ({ src, size = 30 }: Props) => (
```
Comment (evenchange4) [inline packages/mcs-lite-ui/src/Avatar/Avatar.js]: User defaultProps

File: packages/mcs-lite-ui/src/Avatar/Avatar.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      <img alt="" src={src} width={size} height={size} />
+    ) : (
+      <IconAvatar width={size} height={size} />
+    )}
+  </Wrapper>
+);
+
+const enhance = compose(pure);
+const Avatar: React.ComponentType<Props> = enhance(PureAvatar);
+Avatar.displayName = 'Avatar';
```
Comment (evenchange4) [inline packages/mcs-lite-ui/src/Avatar/Avatar.js]: expose the propTypes for the consumer:
```js
```

File: packages/mcs-lite-ui/src/Avatar/Avatar.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  );
+}
+PureAvatar.defaultProps = {
+  size: '30',
+};
+PureAvatar.propTypes = {
+  src: PropTypes.string,
+  size: PropTypes.string,
+};
+
```
Comment (evenchange4) [inline packages/mcs-lite-ui/src/Avatar/Avatar.js]: Avatar.defaultProps = {
  size: '30',
};
Avatar.propTypes = {
  src: PropTypes.string,
  size: PropTypes.string,
};

File: packages/mcs-lite-ui/src/Avatar/styled-components.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+// @flow
+import * as React from 'react';
+import styled from 'styled-components';
+
+const Wrapper: React.ComponentType<*> = styled.div`
+  img,
+  svg {
+    display: inline-block;
+    border-radius: 50%;
+    border: 1px solid ${props => props.theme.color.grayDark};
```
Comment (liuderchi) [inline packages/mcs-lite-ui/src/Avatar/styled-components.js]: NOTE: temporarily use `grayDark` 

(`grayBase` is currently darker than `grayDark`)

https://github.com/MCS-Lite/mcs-lite/blob/8280e339b47042c30ebe3c58184ee741377f31b1/packages/mcs-lite-theme/src/theme.js#L4-L5

File: packages/mcs-lite-ui/src/Avatar/Avatar.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  );
+}
+
+Avatar.displayName = 'Avatar';
+Avatar.defaultProps = {
+  size: 30,
+};
+Avatar.propTypes = {
+  src: PropTypes.string,
+  size: PropTypes.string,
```
Comment (liuderchi) [inline packages/mcs-lite-ui/src/Avatar/Avatar.js]: this should be `PropTypes.number`

File: packages/mcs-lite-ui/src/Avatar/styled-components.js (hunk 4 above)
Comment (evenchange4) [reply packages/mcs-lite-ui/src/Avatar/styled-components.js]: #604

**Pull request merged**